hwmgr-plugins/metal3/controller/baremetalhost_manager.go

```diff
@@ -339,6 +334,24 @@ func clearBMHNetworkData(ctx context.Context, c client.Client, name types.Namesp
 		}
 		return nil
 	})
+	if err != nil {
+		return RequeueAfterShortInterval, fmt.Errorf("failed to clear BMH network data: %w", err)
+	}
+
+	// Wait for metal3 controller to propagate the change to PreprovisioningImage network status
+	networkDataCleared, err := waitForPreprovisioningImageNetworkDataCleared(ctx, c, logger, name)
+	if err != nil {
+		return RequeueAfterShortInterval, fmt.Errorf("failed to check PreprovisioningImage network status for BMH (%s/%s): %w", name.Name, name.Namespace, err)
```

Comment:
If you are returning an error...the interval is ignored https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#TypedReconciler 

this basically saying `return reconcile.Result{}, err ` ...is that what you were looking for? RequeueAfterShortInterval is a bit misleading in that case

Reply:
This is not returning to the reconciler. This is an int being returned to the caller. But yes, those callers currently ignore the first value if an err is returned. The function still has to return something, though.

We'll be reviewing all the various return paths to ensure we're properly requeuing for retriable errors.